Fix DDP docs

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #45476 Fix WorkerInfo link format
* #45458 Update docs for RPC async_execution
* #45455 Fix no_sync docs
* **#45454 Fix DDP docs**

Differential Revision: [D23973367](https://our.internmc.facebook.com/intern/diff/D23973367)

mrshenli committed:
Fix DDP docs

[ghstack-poisoned]

pritamdamania87 commented on torch/nn/parallel/distributed.py:
```diff
@@ -112,37 +112,34 @@ class DistributedDataParallel(Module):
     .. note:: ``nccl`` backend is currently the fastest and
-        highly recommended backend to be used with Multi-Process Single-GPU
-        distributed training and this applies to both single-node and multi-node
-        distributed training
+        highly recommended backend and this applies to both single-node and
+        multi-node distributed training.
```
Should we mention this is our recommended backend for GPU?

pritamdamania87 commented on torch/nn/parallel/distributed.py:
```diff
@@ -208,136 +245,81 @@ class DistributedDataParallel(Module):
+                               unused can be detached from the autograd graph
+                               using ``torch.Tensor.detach``. (default: ``False``)
+        check_reduction: This argument is deprecated.
+        gradient_as_bucket_view (bool): this is a prototype feature and subject
```
nit: Start with caps 'This'

mrshenli committed:
Update on "Fix DDP docs"


Differential Revision: [D23973367](https://our.internmc.facebook.com/intern/diff/D23973367)

[ghstack-poisoned]

mrshenli committed:
Update on "Fix DDP docs"


Differential Revision: [D23973367](https://our.internmc.facebook.com/intern/diff/D23973367)

[ghstack-poisoned]

mrshenli committed:
Update on "Fix DDP docs"


Differential Revision: [D23973367](https://our.internmc.facebook.com/intern/diff/D23973367)

[ghstack-poisoned]

mrshenli committed:
Update on "Fix DDP docs"


Differential Revision: [D23973367](https://our.internmc.facebook.com/intern/diff/D23973367)

[ghstack-poisoned]

mrshenli committed:
Update on "Fix DDP docs"


Differential Revision: [D23973367](https://our.internmc.facebook.com/intern/diff/D23973367)

[ghstack-poisoned]
